app/src/main/java/com/arctouch/codechallenge/viewmodel/MovieListViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.arctouch.codechallenge.viewmodel
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import com.arctouch.codechallenge.api.MoviesRepository
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.NetworkUtils
+import com.arctouch.codechallenge.util.Resource
+
+//Viewmodel destined to all the logic from the MovieListFragment
+class MovieListViewModel : ViewModel() {
+    //data repository
+    private lateinit var repository: MoviesRepository
+
+    //Page number for pagination
+    private var pageNumber: Long = 1
+
+    val moviesData = MutableLiveData<Resource<ArrayList<Movie>>>()
+
+    //init repository
+    fun init(){
+        repository = MoviesRepository(NetworkUtils.getRetrofitInstance())
+    }
+
+    //setup movies data
+    fun getMoviesData(refreshValues: Boolean = false){
+        moviesData.value = Resource.loading()
+
+        //refresh list back to page 1
+        if(refreshValues) pageNumber = 1
+
+        //detect if has genre already cached to avoid network calls
+        if(Cache.hasGenreCached()) getMovieList()
+        else getGenre()
```

Comment:
The formatting you're using for `if`s and `when`s  is unusual. Is this a standard pattern you've used on previous projects? Would you be open to using different coding formats?

Reply:
yes, it's a pattern that i found to use in my current job with my team, but of course I'm totally open to another coding format and patterns!